src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,625 @@
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private HttpClient httpClient;
+
+    @Mock
+    private Clock clock;
+
+    @Mock
+    private PrebidVersionProvider prebidVersionProvider;
+
+    @Captor
+    private ArgumentCaptor<String> jsonCaptor;
+
+    @Captor
+    private ArgumentCaptor<MultiMap> headersCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> urlCaptor;
+
+    @Captor
+    private ArgumentCaptor<Long> timeoutCaptor;
+
+    private GreenbidsAnalyticsReporter target;
+
+    private GreenbidsAnalyticsProperties greenbidsAnalyticsProperties;
+
+    @Before
+    public void setUp() {
+        greenbidsAnalyticsProperties = GreenbidsAnalyticsProperties.builder()
+                .exploratorySamplingSplit(0.9)
+                .analyticsServerVersion("2.2.0")
+                .analyticsServerUrl("http://localhost:8080")
+                .configurationRefreshDelayMs(10000L)
+                .timeoutMs(100000L)
+                .build();
+
+        target = new GreenbidsAnalyticsReporter(
+                greenbidsAnalyticsProperties,
+                jacksonMapper,
+                httpClient,
+                clock,
+                prebidVersionProvider);
+
+        jsonCaptor = ArgumentCaptor.forClass(String.class);
+        headersCaptor = ArgumentCaptor.forClass(MultiMap.class);
+        urlCaptor = ArgumentCaptor.forClass(String.class);
+        timeoutCaptor = ArgumentCaptor.forClass(Long.class);
+    }
+
+    @Test
+    public void shouldReceiveValidResponseOnAuctionContextForBanner() throws IOException {
+        // given
+        final Banner banner = givenBanner();
+
+        final ObjectNode prebidJsonNodes = mapper.valueToTree(
+                singletonMap("gpid", TextNode.valueOf("gpidvalue")));
+
+        final Imp imp = Imp.builder()
+                .id("adunitcodevalue")
+                .ext(prebidJsonNodes)
+                .banner(banner)
+                .build();
+
+        final AuctionContext auctionContext = givenAuctionContext(imp);
+        final AuctionEvent event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        final HttpClientResponse mockResponse = mock(HttpClientResponse.class);
+        when(mockResponse.getStatusCode()).thenReturn(202);
+        when(httpClient.post(anyString(), any(MultiMap.class), anyString(), anyLong()))
+                .thenReturn(Future.succeededFuture(mockResponse));
+        final CommonMessage expectedCommonMessage = givenCommonMessageForBanner();
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        // then
+        assertThat(result.succeeded()).isTrue();
+        verify(httpClient).post(
+                urlCaptor.capture(), headersCaptor.capture(), jsonCaptor.capture(), timeoutCaptor.capture());
```

Comment:
there is a static predefined captors in the mockito, so if it's a single value, you can just use the `eq(...)` equality captor (check other unit test for more examples)

`eq` also can compare objects, so you can create a map of expected headers and use `eq` for the map

so please make it simpler

Reply:
As discussed I've added the mockito `eq` for URL, timeout. For headers check kept the captor capture